File: design/src/commonMain/kotlin/dev/helw/playground/sdui/design/core/IconToken.kt
package dev.helw.playground.sdui.design.core

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.ArrowForward
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.List
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Place
import androidx.compose.material.icons.filled.Settings
import androidx.compose.ui.graphics.vector.ImageVector

enum class IconToken(val image: ImageVector) {
    HOME(Icons.Default.Home),
    ADD(Icons.Default.Add),
    DELETE(Icons.Default.Delete),
    FAVORITE(Icons.Default.Favorite),
    SETTINGS(Icons.Default.Settings),
    LIST(Icons.Default.List),
    CHECK(Icons.Default.Check),
    ARROW_FORWARD(Icons.Default.ArrowForward),
    ARROW_BACK(Icons.Default.ArrowBack),
    PLACE(Icons.Default.Place),
    INFO(Icons.Default.Info),
}

File: catalog-ios/Design/Core/IconToken.swift
//
//  IconToken.swift
//  Catalog
//
//  Created by Ahmed El-Helw on 9/9/23.
//

import Foundation

enum IconToken : String, Codable, CaseIterable {
  case home
  case add
  case delete
  case favorite
  case settings
  case list
  case check
  case arrowForward
  case arrowBackward
  case place
  case info
}

extension IconToken {
  var value: String {
    switch self {
      case .home:
        "house.fill"
      case .add:
        "plus"
      case .delete:
        "trash.fill"
      case .favorite:
        "heart.fill"
      case .settings:
        "gearshape.fill"
      case .list:
        "list.bullet"
      case .check:
        "checkmark"
      case .arrowForward:
        "arrow.forward"
      case .arrowBackward:
        "arrow.backward"
      case .place:
        "mappin"
      case .info:
        "info.circle.fill"
    }
  }
}
